### PR TITLE
Validate leankit on pr 556125341

### PR DESCRIPTION
**Pull request by ilyakatz**
This add validation for presence of leankit in PR commits

image

### CODE REVIEW

File: deployer.rb [hunk 1]
```diff
@@ -70,8 +71,14 @@ def client
...
 
-  def leankit_ticket_present?(branch_name)
-    !!(branch_name =~ /(\d{6,10})/)
+  def leankit_ticket_present?(pull_request)
+    commit_messages(pull_request).any? do |comment|
+      leankit?(comment)
+    end
+  end
+
+  def leankit?(comment)
```
Comment (joegesualdo) [inline deployer.rb]: I read this as "is this a leankit comment". Maybe is should read more like "is a leankit card identifier present in the comment".

File: deployer.rb [hunk 2]
```diff
@@ -70,8 +71,14 @@ def client
...
+    !!(comment =~ /(\d{6,10})/)
```
Comment (joegesualdo) [inline deployer.rb]: What is this doing, exactly?

File: deployer.rb [hunk 3]
```diff
@@ -41,26 +41,27 @@ def process_circle_ci(payload)
   end
 
   def process_pull_request(pull_request)
```
Comment (joegesualdo) [inline deployer.rb]: You might be able to clean up the duplication here.

For example, if you changed the `hotfix?` method to take a `pull_request` object, like the `leankit_ticket_present?`, then you could have something like this:

```ruby
description = 
  if leankit_ticket_present?(pull_request)
    hotfix?(pull_request) ? "This is a hotfix!" :  slack_channel_topic
  else
    "None of the messages in the PR include LeanKit ID"

status = 
  leankit_ticket_present?(pull_request) || hotfix?(pull_request) 
    ? build_status 
    : "error",

client.create_status(
  repo_name(pull_request),
  pull_request['head']['sha'],
  status,
  opts.merge(description: description)
)
```

File: deployer.rb [hunk 4]
```diff
@@ -108,4 +119,16 @@ def slack_client
     @slack_client = Slack::Client.new
   end
 
+  def commit_messages(pull_request)
+    name = repo_name(pull_request)
+    number = pull_request_number(pull_request)
+    commits = client.pull_request_commits(name, number)
+    commits.collect do |c|
```
Comment (joegesualdo) [inline deployer.rb:118]: I had to lookup the difference between `collect` and `map`... turns out they're the same, haha

File: deployer.rb (hunk 3 above)
Comment (joegesualdo) [reply deployer.rb]: And normally when I start creating variables in a method, I think "could this be a reusable function"?

```ruby
# description
# Get the description for a pull request
# 
# description :: PullRequest -> String
def description(pull_request)
  if leankit_ticket_present?(pull_request)
    hotfix?(pull_request) ? "This is a hotfix!" :  slack_channel_topic
  else
    "None of the messages in the PR include LeanKit ID"

# status
# Get the status for a pull request
#
# status :: PullRequest -> String
def status(pull_request) 
  leankit_ticket_present?(pull_request) || hotfix?(pull_request) 
    ? build_status 
    : "error"
end
```

File: deployer.rb (hunk 1 above)
Comment (ilyakatz) [reply deployer.rb]: changed to `leankit_identifiable`

File: deployer.rb (hunk 2 above)
Comment (ilyakatz) [reply deployer.rb]: this is poor man's identification of leankit card, just checks if there is a long number. i don't think there is any other way to identify what leankit id is

File: deployer.rb [hunk 5]
```diff
@@ -82,8 +72,21 @@ def repo_name(pull_request)
...
+  def build_status(pull_request)
+    leankit_ticket_present?(pull_request) ? "success" : "error"
+  end
+
+  def description(pull_request)
+    d = leankit_ticket_present?(pull_request) ?
+      slack_channel_topic :
+      "None of the messages in the PR include LeanKit ID"
+
+    puts d
```
Comment (joegesualdo) [inline deployer.rb:88]: Did you mean for this to print to be here?

Comment (ilyakatz) [reply deployer.rb:88]: Yeah, i want to keep it there for debugging purproses so i have something to look for in the logs associated with API calls

**Review by joegesualdo — APPROVED** [review]

**Pull request merged**